sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/WindowedSubscriber.java

```diff
@@ -801,19 +790,23 @@ private void terminate(WorkTerminalState terminalState) {
                 timers.dispose();
             } finally {
                 if (terminalState == WorkTerminalState.SINK_ERROR) {
-                    withPendingKey(logger.atWarning()).log(TERMINATING_WORK, "(Reason: sink-error)");
+                    withPendingKey(logger.atWarning())
+                        .addKeyValue("reason", "sink-error")
+                        .log(TERMINATING_WORK);
```

Comment:
it allows to query all messages with something like message == "terminating work" and reason == "sink-error" instead of doing unreliable string parsing

Reply:
nice, thanks Liudmila! lgtm